Pass the info around correctly when describing/evaluating

Drup opened the pull request:
Now, you really only need to implement `Graph.pp`. :]
Also, please put back `Key.Set` in the API ...

Drup commented:
Hum, it seems I reindented by mistake and I don't have the same ocp-indent options than you do. Can you commit the .ocp-indent that you want please ?

Drup commented:
ping.

samoht commented:
> Hum, it seems I reindented by mistake and I don't have the same ocp-indent options than you do. Can you commit the .ocp-indent that you want please ?

I'm using the default parameters, weird. I've just seen this PR, I'll rebase on top of my less optimal changes.

Drup commented:
> I'm using the default parameters, weird.

Yes, but I don't. :p

samoht commented:
> Now, you really only need to implement Graph.pp. :]

How do you do this with your changes? I need the info to get project name, libs and packages

Drup commented:
You conflated printing info and printing the graph. I just made them separate again.

samoht commented:
well, I just want `mirage describe` to print info about the current project...

Drup commented on app/functoria_app.ml:
```diff
@@ -579,14 +573,15 @@ module Make (P: S) = struct
-        show_keys map @@ Key.deps info;
-        Key.eval map info @@ map
+        let e = Key.eval map info @@ map in
+        Log.info "@[<v>%a@]" (Info.pp false) (snd e) ;
```
This prints info about the project (regardless which command is used). You added verbosity info so you can put that in there.

Drup commented:
I just added unconditional printing of the `Info.t` after evaluation. It's going to be printed with all the command (which actually also improved some printing that were done before for `mirage configure`).

In order not to flood the output for other command, I added a verbosity flag for the `Info.pp` command, but didn't implemented the command line flag, which you did. :p
Either we turn this verbosity up by default for `describe` or we don't, your choice.

Is that clearer ?

samoht commented:
yep that's clearer thanks. I'll turn on verbosity when the flag is set.

Drup commented:
Good! Sorry for the monosyllabic explanation before. :p

samoht commented:
Can I print the info in the `describe` command a bit above? Not sure to fully understand when things are fully evaluated or not. I guess the describe takes the evaluated graph and info as argument.

Drup commented:
`evaluated` is a pair of the info (including the context) and a graph. I'm not sure I understand why you want exactly ? "above" what ?

samoht commented:
See https://github.com/samoht/functoria/commit/bd0635ba4923a7ad20155867b52c0f79183cb95c

Drup commented:
That looks good.

Drup commented:
This has been merged manually.
